scripts/combine/setupCombine.py

```diff
@@ -409,107 +411,125 @@ def assertSample(name, startsWith=["W", "Z"], excludeMatch=[]):
-                                    preOpMap={
-                                        m.name: (lambda h: hh.addHists(h[{ax: hist.tag.Slicer()[::hist.sum] for ax in poi_axes}],
-                                                                        h[{"ptVgenSig": hist.tag.Slicer()[0:h.axes["ptVgenSig"].size:hist.sum],
-                                                                            "absYVgenSig": hist.tag.Slicer()[hist.overflow:]}],
-                                                                        scale2=args.scaleNormXsecHistYields)
-                                                    ) if sign in m.name else (lambda h: h[{ax: hist.tag.Slicer()[::hist.sum] for ax in poi_axes}]) for g in cardTool.procGroups["signal_samples"] for m in cardTool.datagroups.groups[g].members
-                                    },
-                                    )
-                # ptV OOA and yV OOA, integrate helicities
-                cardTool.addSystematic("yieldsTheoryAgnostic",
-                                    rename=f"yieldsTheoryAgnostic_OOA_ptVyV_{label}{sign}",
-                                    **noi_args,
-                                    mirror=True,
-                                    scale=0.5,
-                                    processes=["signal_samples"],
-                                    baseName=f"{nuisanceBaseNameOOA}{sign}_",
-                                    noConstraint=True if args.priorNormXsec < 0 else False,
-                                    systAxes=["ptVgenSig", "absYVgenSig","helicitySig"],
-                                    formatWithValue=[None,None,"low"],
-                                    labelsByAxis=["PtVBin", "YVBin", "Helicity"],
-                                    systAxesFlow=["ptVgenSig", "absYVgenSig"], # this can activate nuisances on overflow bins, mainly just ptV and yV since the helicity axis has no overflow bins
-                                    skipEntries=[{"helicitySig" : [6,7,8]}], # removing last three indices out of 9 (0,1,...,7,8) corresponding to A5,6,7
-                                    preOpMap={
-                                        m.name: (lambda h: hh.addHists(h[{ax: hist.tag.Slicer()[::hist.sum] for ax in poi_axes}],
-                                                                        h[{"ptVgenSig": hist.tag.Slicer()[hist.overflow:],
-                                                                            "absYVgenSig": hist.tag.Slicer()[hist.overflow:]}],
-                                                                        scale2=args.scaleNormXsecHistYields)
-                                                    ) if sign in m.name else (lambda h: h[{ax: hist.tag.Slicer()[::hist.sum] for ax in poi_axes}]) for g in cardTool.procGroups["signal_samples"] for m in cardTool.datagroups.groups[g].members
-                                    },
-                                    )
-
+            if args.theoryAgnosticPolVar:
```

Comment:
Do you mean the whole theory agnostic part including the one with the norm vars that we had before? 
I am iffy, I personally prefer that I open setupCombine and see what's going on rather than having to browse other files just to know what systematics I am using. It was making sense for pdfs and QCD scales because there were a lot of operations ongoing before the call to addSystematics

Reply:
Well, the point is that if you aren't using theoryAgnosticPolVar, you don't care at all what is happening in these 100 lines, so it makes sense for it to be in another file. Just call a function with a clear name, and if someone cares what's going on, they can check that function. If you don't care you just trust that the function does what it says

Reply:
Ok, so maybe what I did was more than you had in mind, now both theory agnostic versions are put into a different script with a class handling the cardtool.addsystematics calls, but I guess it is still fine